connectors/sources/confluence.py

```diff
@@ -0,0 +1,261 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Confluence source module responsible to fetch documents from Confluence Cloud/Server.
+"""
+import os
+
+import aiohttp
+from aiohttp.client_exceptions import ServerDisconnectedError
+
+from connectors.logger import logger
+from connectors.source import BaseDataSource
+from connectors.utils import CancellableSleeps, iso_utc, ssl_context
+
+FILE_SIZE_LIMIT = 10485760
+RETRY_INTERVAL = 2
+PING_URL = "rest/api/space?limit=1"
+MAX_CONCURRENT_DOWNLOADS = 50  # Max concurrent download supported by confluence
+
+
+class ConfluenceClient:
+    """Confluence client to handle API calls made to Confluence"""
+
+    def __init__(self, configuration):
+        self._sleeps = CancellableSleeps()
+        self.configuration = configuration
+        self.is_cloud = self.configuration["is_cloud"]
+        self.host_url = self.configuration["host_url"]
+        self.ssl_enabled = self.configuration["ssl_enabled"]
+        self.certificate = self.configuration["ssl_ca"]
+        self.retry_count = self.configuration["retry_count"]
+        if self.is_cloud:
+            self.host_url = os.path.join(self.host_url, "wiki")
+
+        if self.ssl_enabled and self.certificate:
+            self.ssl_ctx = ssl_context(certificate=self.certificate)
+        else:
+            self.ssl_ctx = False
+        self.session = None
+
+    def get_session(self):
```

Comment:
We should probably add a leading underscore here as it's an "internal" method.

Reply:
done [16f175d](https://github.com/elastic/connectors-python/pull/568/commits/16f175d3b6bb1684cee74494699333fb82fabd11)